tests/test_main.py

```diff
@@ -162,7 +163,7 @@ def test_vr_run_miss_model(mocker, caplog):
     mock_conf.return_value = {"core": {"modules": ["topsoil"]}}
 
     with pytest.raises(InitialisationError):
-        vr_run("tests/fixtures/all_config.toml", ".", "delete_me")
+        vr_run("tests/fixtures/all_config.toml", Path("./delete_me.toml"))
 
```

Comment:
What's this `delete_me.toml` file? Is it a real file? I don't understand how it is being used.

Reply:
It's a file path to write the complete configuration to, if it exists at the start then the test would fail, which is part of why I've given it a stupid name. The other reason I gave it a stupid name is that if for some reason the test doesn't fail and raise an `InitialisationError`, `vr_run` will actually write the config to that name. So I gave it that name during writing the unit tests to remind me to delete it when it got accidentally written out

Reply:
Got it. That seems clunky - we don't want tests that pollute the file space on failure. I'm not sure of the best solution though - the obvious one is to use `pyfakefs` to set up a virtual file system to run tests in. That seems like a sledgehammer to crack this, but we might well make use of it later.

Reply:
Alternatives:

 -  the test could look for the file and delete it on the way out.
 - you could write to an explicit [tempfile](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp), but that is just moving the litter to somewhere we can't see it 😄